src/components/navigation/navigation.tsx

```diff
@@ -0,0 +1,201 @@
+import * as React from 'react';
+import Image from 'next/image';
+import * as NavigationMenuPrimitive from '@radix-ui/react-navigation-menu';
+import * as AccordionPrimitive from '@radix-ui/react-accordion';
+import { cva, type VariantProps } from 'class-variance-authority';
+import { ChevronDownIcon } from '@stash-ui/regular-icons';
+const subItem1SVG = require('@/assets/images/sub-item1.svg');
+const subItem2SVG = require('@/assets/images/sub-item2.svg');
+const subItem3SVG = require('@/assets/images/sub-item3.svg');
+const subItem4SVG = require('@/assets/images/sub-item4.svg');
+
+import { cn } from '@/lib/utils';
+
+const navigationVariants = cva(
+  'inline-flex justify-between w-full p-3 rounded-lg cursor-pointer text-sm font-normal',
+  {
+    variants: {
+      variant: {
+        highlighted:
+          'shadow shadow-[0px_0px_0px_3px_rgba(14,159,110,0.12)] hover:bg-list-highlighted text-success-foreground active:font-semibold',
+        default:
+          'text-tertiary-foreground hover:text-active-foreground hover:bg-list-actived active:font-semibold active:bg-active-menu active:text-active-foreground',
+        disabled: 'text-tertiary-foreground opacity-45 cursor-not-allowed',
+      },
+    },
+    defaultVariants: {
+      variant: 'default',
+    },
+  }
+);
+
+export interface NavigationLinkProps
+  extends React.ComponentProps<typeof NavigationMenuPrimitive.Link>,
+    VariantProps<typeof navigationVariants> {}
+
+function NavigationLink({ className, variant, ...props }: NavigationLinkProps) {
+  return (
+    <NavigationMenuPrimitive.Root>
+      <NavigationMenuPrimitive.Link
+        data-testid='navigation-link'
+        className={cn(navigationVariants({ variant }), className)}
+        {...props}
+      />
+    </NavigationMenuPrimitive.Root>
+  );
+}
+
+export interface NavigationItemProps
+  extends React.ComponentProps<typeof NavigationMenuPrimitive.Item>,
+    VariantProps<typeof navigationVariants> {}
+
+function NavigationItem({ className, variant, ...props }: NavigationItemProps) {
+  const [isActive, setIsActive] = React.useState(false);
+
+  return (
+    <NavigationMenuPrimitive.Root>
+      <NavigationMenuPrimitive.Item
+        data-testid='navigation-item'
+        className={cn(
+          navigationVariants({ variant }),
+          className,
+          isActive &&
+            !variant &&
+            'text-active-foreground bg-active-menu font-semibold',
+          isActive &&
+            variant === 'highlighted' &&
+            'text-success-foreground bg-list-highlighted font-semibold'
+        )}
+        onClick={() => setIsActive(!isActive)}
```

Comment:
what will happen if someone passes a custom `onClick` to this component? You need to create a `handleClick` that calls onClick via props and also call `setIsActive`